### PR TITLE
A couple more fixes cuz why not!

### PR DESCRIPTION
**Pull request by fat-tire**
Thanks for the pull!  You did such a great batch of changes so fast--- so I felt like I should do more!

 I worked a little on the UI like numerals only in some fields, a multiline prompt, the saving-settings feature (will auto-load if the file is there too), disable start when its running, autoscroll the log to the bottom, fixed a few other minor bugs, simplified the code in a few places, plus added a "random seed every time" checkbox.  If you try to enter a height or width not a multiple of 64 it'll fix it for you too -- though it will accept small sizes like 0 and 64 :)  Could still have it auto-save settings whenever you make a change or when you quit, but I didn't go that far.

Hope this is okay.  Some of these individual commits may not quite standalone as I did it all at once then tried to break it into bite-sized commits and maybe screwed up- not sure.  But they should work together.  You may need to kill your settings.json file to accommodate the slightly new format.

Looks like this now:

image

ft

*thanks to @Fictiverse for inspiring the prompt on Twitter

### CODE REVIEW

**Comment by fat-tire** [comment]
I should add it now also saves the prompt-- so with the seed and the prompt and the dimensions and everything I think any image could be reproduced from just the `settings.json` file (?)

**Comment by 0xBLCKLPTN** [comment]
Hi, I would like to sincerely thank you for this tremendous work! For about a week I will hardly be able to develop this repository further because of my main work. I thought that if you can keep adding new features yourself, I should add you as a collaborator. Thanks again!

File: neuro-gui.py [hunk 1]
```diff
@@ -224,71 +255,71 @@ def sel_dir(self):
...
             json.dump(res, outfile)
         logging.info("Settings saved!")
         print(res)
 
     def plms_func(self, state):
-        if state == QtCore.Qt.Checked:
-            self.plms = True
-        else:
-            self.plms = False
+        self.plms = state == QtCore.Qt.Checked
```
Comment (0xBLCKLPTN) [inline neuro-gui.py:319]: perfect;)

File: neuro-gui.py [hunk 2]
```diff
@@ -156,23 +182,35 @@ def _setMainUi(self):
         self.setCentralWidget(self.widget)
 
     def log_subprocess_output(self, pipe):
```
Comment (0xBLCKLPTN) [inline neuro-gui.py:184]: This function was only for the test and it is probably not needed. I used it to get the output from the image generator into the debug window. But it didn't work

**Review by 0xBLCKLPTN — COMMENTED** [review]
You've done a great job. Soon, I'm going to change the style for the program using ``css``.
I sent you an invite to the collaborators ;)

**Pull request merged**